src/functions.php

```diff
@@ -1,21 +1,22 @@
 <?php
 
-function produce_HTML_Head(): string {
-	return '<html lang="en-gb"><head><meta charset="utf-8"><title>Cosmic Collector</title>
-	<link rel="stylesheet" type="text/css" href="normalize.css">
-	<link rel="stylesheet" type="text/css" href="cosmic_collector.css"></head>';
-}
-
-
-function display_Cosmic_Event(array $db_row): string {
-	return '<div class="cosmic_event"><h2>' . $db_row['name'] . '</h2><h3>Caused by: ' . $db_row['caused_by']
-		. '</h3><h3>Creates: ' . $db_row['creates_effect'] . '</h3><h3>Produces: </h3></div>';
+function format_event_elements(array $elements): array {
+	$elements_produced = [];
+	foreach ($elements as $element){
+		if (array_key_exists('name', $element)) $elements_produced[] = $element['name'];
+		else return ['FAIL'];
+	}
+	return $elements_produced;
 }
 
-function extract_Event_Elements(array $db_query_results): array {
-	$event_elements = [];
-	foreach ($db_query_results as $row){
-		$event_elements[] = $row['name'] . '-' . $row['atomic-mass'];
+function display_cosmic_event(array $db_row, array $event_elements): string {
+	if (!array_key_exists('name', $db_row) || !array_key_exists('caused_by', $db_row) ||
+		!array_key_exists('creates_effect', $db_row) || count($event_elements) < 1)
+		return 'FAIL';
+	$html = '<div class="cosmic_event"><h2>' . $db_row['name'] . '</h2><h3>Caused by: ' . $db_row['caused_by']
+		. '</h3><h3>Creates: ' . $db_row['creates_effect'] . '</h3><h3>Produces:</h3><p>';
+	foreach ($event_elements as $element){
+		$html .= $element . ', ';
```

Comment:
there's a comma at the end of the list - ideally it should be without the comma
![image](https://user-images.githubusercontent.com/25844355/128187225-7b1f2d74-7b13-402c-aba6-cc7794b42b45.png)